ngraph/src/ngraph/op/split.cpp

```diff
@@ -186,3 +190,94 @@ shared_ptr<Node> op::v1::Split::clone_with_new_inputs(const OutputVector& new_ar
     check_new_args_count(this, new_args);
     return make_shared<v1::Split>(new_args.at(0), new_args.at(1), m_num_splits);
 }
+
+namespace
+{
+    template <element::Type_t ET>
+    inline bool evaluate(const HostTensorPtr& in,
+                         const HostTensorPtr& out,
+                         const Coordinate& lower_bounds,
+                         const Coordinate& upper_bounds)
+    {
+        runtime::reference::slice(in->get_data_ptr<ET>(),
+                                  out->get_data_ptr<ET>(),
+                                  in->get_shape(),
+                                  lower_bounds,
+                                  upper_bounds,
+                                  Strides(lower_bounds.size(), 1),
+                                  out->get_shape());
+
+        return true;
+    }
+
+    bool evaluate_split(const HostTensorPtr& data_tensor,
+                        const HostTensorPtr& axis_tensor,
+                        const HostTensorVector& outputs,
+                        const int64_t num_splits,
+                        const Node* split_node)
+    {
+        bool rc = true;
+
+        const auto axis = read_vector<int64_t>(axis_tensor)[0];
+        const auto normalized_axis =
+            ngraph::normalize_axis(split_node, axis, data_tensor->get_partial_shape().rank());
+        const auto data_shape = data_tensor->get_shape();
+        const size_t axis_dim_length = data_shape.at(normalized_axis);
+        const size_t part_length = axis_dim_length / num_splits;
+
+        Shape output_shape = data_shape;
+        output_shape.at(normalized_axis) = part_length;
+
+        std::vector<size_t> lower_bounds(data_shape.size(), 0);
+        std::vector<size_t> upper_bounds = data_shape;
+        upper_bounds.at(normalized_axis) = part_length;
+
+        for (const auto& output : outputs)
+        {
+            output->set_shape(output_shape);
+            switch (data_tensor->get_element_type())
+            {
+                TYPE_CASE(i8)(data_tensor, output, lower_bounds, upper_bounds);
```

Comment:
Can we reduce the number of supported data types? for concat we have the next list:
```
            TYPE_CASE(i32)(args, out, concatenation_axis);
            break;
            TYPE_CASE(i64)(args, out, concatenation_axis);
            break;
            TYPE_CASE(u32)(args, out, concatenation_axis);
            break;
            TYPE_CASE(u64)(args, out, concatenation_axis);
            break;
            TYPE_CASE(f16)(args, out, concatenation_axis);
            break;
            TYPE_CASE(f32)(args, out, concatenation_axis);
            break;
```

For example at the current moment OV doesn't support f64